ormolu-live/cbits/init.c

```diff
@@ -1,14 +1,28 @@
+#include "Rts.h"
+
 #include "Main_stub.h"
-#include <Rts.h>
+
+void malloc_inspect_all(void (*handler)(void *start, void *end,
+                                        size_t used_bytes, void *callback_arg),
+                        void *arg);
+
+static void malloc_inspect_all_handler(void *start, void *end,
+                                       size_t used_bytes, void *callback_arg) {
+  if (used_bytes == 0) {
+    memset(start, 0, (size_t)end - (size_t)start);
+  }
+}
```

Comment:
Maybe add a link to eg https://github.com/WebAssembly/wasi-libc/blob/be1704a9568e98ae073f074f7271cae68c23d161/dlmalloc/src/malloc.c#L1027-L1052 (or is there a better reference)? But I guess you might add this trick to [ghc-wasm-meta](https://gitlab.haskell.org/ghc/ghc-wasm-meta/), in which case it doesn't need to be explained here.

Reply:
Yes, I'll add this to `ghc-wasm-meta` documentation with more explanation. `malloc_inspect_all` is a `dlmalloc` internal function to traverse the entire heap, not exposed by upstream `wasi-libc`, but I enabled it in our fork for this purpose.